internal/controller/component_customizer.go

```diff
@@ -257,8 +257,8 @@ func customizeContainer(cSpec operatorv1.ContainerSpec, d *appsv1.Deployment) {
 				c.Resources = *cSpec.Resources
 			}
 
-			if cSpec.Image != nil && cSpec.Image.Name != "" && cSpec.Image.Repository != "" {
-				c.Image = imageMetaToURL(cSpec.Image)
```

Comment:
how are these changes related to new API version?

Reply:
I guess this is the reason for the new API version. Incompatible fields

Reply:
Yes, @Danil-Grigorev is right. It's a result of the API change. In v1alpha1 we used ImageMeta object to provide a custom image for the provider. In v1alpha2 we decided that it's better to use the well-known <registry>/<name>:<tag> image URL format, which is just a string.
After this change we don't have to convert ImageMeta -> ImageURL anymore, hence the change.

Reply:
Should we add documentation for user migration from v1alpha1 to v1alpha2 in this case? Similar to https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/providers/version-migration.md?plain=1

Reply:
But migration doc in CAPI is formed when they release a minor release, and describe changes needed going from one minor release of CAPI to another.

Reply:
Technically, you don't need to migrate anything manually. I implemented auto-conversion, which handles v1alpha1 -> v1alpha2 upgrade.
But I agree that it would be nice to have a doc describing the changes in the new version.